### ISSUE TITLE
react-leaflet-deflate not compatible with react-leaflet v4

### ISSUE THREAD

**Issue by hdale94** [mhasbie/react-leaflet-deflate#8]
Hi @mhasbie @oliverroick ,

I can't get this package to work in react-leaflet v4 and was wondering if anyone has any solutions. 

Here's the error:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: plokk-report-app@0.1.0
npm ERR! Found: react-leaflet@4.1.0
npm ERR! node_modules/react-leaflet
npm ERR!   react-leaflet@"^4.0.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react-leaflet@"^1.9.1 || ^2.1.4" from react-leaflet-deflate@2.2.1
npm ERR! node_modules/react-leaflet-deflate
npm ERR!   react-leaflet-deflate@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

I tried installing with "--legacy-peer-deps", but got this error when testing an example from the README.md:

`Super expression must either be null or a function, not undefined`

**Comment by XanderSelorm** [comment]
@hdale94 , have you found a way around it?

**Comment by hdale94** [comment]
No I did not. 

This one could be useful though:
[https://github.com/changey/react-leaflet-markercluster](url)

**Comment by XanderSelorm** [comment]
Thank you for the package. But can I use it to cluster polygons? It doesn't work for me. Maybe I'm using it wrong.

**Comment by hdale94** [comment]
No, you can not cluster polygons with that one either, unfortunately. 

My suggested solution would be to use a marker with a popup, and inside the popup, you can display a small leaflet window with the polygon. You can customize the marker icon to indicate that it contains a polygon.

**Comment by XanderSelorm** [comment]
> No, you can not cluster polygons with that one either, unfortunately.
> 
> My suggested solution would be to use a marker with a popup, and inside the popup, you can display a small leaflet window with the polygon. You can customize the marker icon to indicate that it contains a polygon.

Thanks again for replying. I guess I'll have to try your suggestion too and see how the outcome may be. 
Cheers!

**Comment by mhasbie** [comment]
@hdale94 , this issue is now fixed in v3.0.0. Thanks for reporting.
